src/bin/stratis-min/main.rs

```diff
@@ -46,13 +49,18 @@ fn main() -> Result<(), String> {
     if let Some(subcommand) = args.subcommand_matches("key") {
         if let Some(args) = subcommand.subcommand_matches("set") {
             key::key_set(
-                args.value_of("key_desc").expect("required"),
+                &KeyDescription::try_from(args.value_of("key_desc").expect("required").to_owned())
+                    .expect("try_from"),
```

Comment:
@mulkieran What message should I print here? Is there a better way to handle errors that may arise in try_from?

Reply:
@lleshchi Yes. You only want to use expect() if you can prove it must succeed. The argument for the expect() on value_of() is that the parser has been set up so that it ensures that a value for "key_desc" is always available, and so that can never fail. But the user could pass an invalid string for the key description, our code can't prevent that.

The thing to look at is the return type of the method, ```Result<(), String>```. When you return an error, it must be in the form of a String. There's examples in the method itself, using ```map_err```, to convert another kind of error to a String. Try following that exampel.